example/points/generate_points.c

```diff
@@ -0,0 +1,246 @@
+/*
+  This file is part of p4est.
+  p4est is a C library to manage a collection (a forest) of multiple
+  connected adaptive quadtrees or octrees in parallel.
+
+  Copyright (C) 2010 The University of Texas System
+  Additional copyright (C) 2011 individual authors
+  Written by Carsten Burstedde, Lucas C. Wilcox, and Tobin Isaac
+
+  p4est is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2 of the License, or
+  (at your option) any later version.
+
+  p4est is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with p4est; if not, write to the Free Software Foundation, Inc.,
+  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+*/
+
+#ifdef THREED
+#include <p4est_to_p8est.h>
+#endif
+
+#ifdef P4_TO_P8
+#include <p8est_bits.h>
+#include <p8est_points.h>
+#include <p8est_vtk.h>
+#else
+#include <p4est_bits.h>
+#include <p4est_points.h>
+#include <p4est_vtk.h>
+#endif /* !P4_TO_P8 */
+#include <sc_io.h>
+
+
+static void
+generate_points (const char *filename,
```

Comment:
Will this file compile when MPIIO is not enabled (try --disable-mpiio)? This relates to libsc, where MPI_File_get_view etc. are defined with MPIIO, but have no correspondence without it.

Reply:
ok, MPIIO is required. We can disable building this example when mpi-io is not available.

Reply:
This seems fine for now. @tim-griesbach do you see a way to abstract the MPI I/O status away in the future?

Reply:
We can abstract the MPI I/O status away using the functions `sc_io_open`, `sc_io_write_at_all`, `sc_io_read_at_all` and `sc_io_close` since these function work also without MPI I/O or even without MPI.

However, in `points2.c` the number of points on reading is currently determined using the file size. Up to my understanding this does not work without MPI I/O in a general portable way by just relying on the C standard (cf. https://wiki.sei.cmu.edu/confluence/display/c/FIO19-C.+Do+not+use+fseek()+and+ftell()+to+compute+the+size+of+a+regular+file). 

We can solve this problem by writing the number of points to the file. Assuming this adjustment we shall obtain code that produces equivalent files with and without MPI I/O.

Reply:
If you want, I can adjust the code as described above.